tests/dbt/parser/test_project.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+
+from cosmos.providers.dbt.parser.project import DbtModel, DbtModelConfig, DbtProject
+
+
+def test_dbt_model_config_selectors_add():
+    """Ensure that materialized and schema are not updated once set but tags are appended."""
+    sql_config = DbtModelConfig(
+        config_selectors={"materialized:incremental", "tags:finance"}
+    )
+    schema_config = DbtModelConfig(
+        config_selectors={"materialized:table", "schema:jaffle_shop", "tags:hourly"}
+    )
+    project_config = DbtModelConfig(
+        config_selectors={
+            "materialized:view",
+            "schema:my_project",
+            "tags: confidential",
+        }
+    )
+    merged_config = sql_config + schema_config + project_config
+    assert merged_config.config_selectors == {
+        "materialized:incremental",
+        "schema:jaffle_shop",
+        "tags:finance",
+        "tags:hourly",
+        "tags: confidential",
+    }
+
+
+def test_dbt_model_multi_tag(tmp_path: Path):
+    sql_model = tmp_path.joinpath("dummy.sql")
+    sql_query = """
+    {{
+        config(
+            materialized='incremental',
+            partition_by='organisation',
+            schema='jaffle_shop',
+            tags=['finance', 'daily', 'confidential']
+        )
+    }}
+    SELECT *
+    FROM {{ ref('my_upstream_model') }}
+    """
+    sql_model.write_text(sql_query)
+    dbt_model = DbtModel("jaffle_shop_revenue", sql_model)
+    dbt_model_config = dbt_model.config
+    assert dbt_model_config.config_selectors == {
+        "materialized:incremental",
+        "schema:jaffle_shop",
+        "tags:finance",
+        "tags:daily",
+        "tags:confidential",
+    }
+
+
+def test_dbt_model_single_tag(tmp_path: Path):
+    sql_model = tmp_path.joinpath("dummy.sql")
+    sql_query = """
+    {{
+        config(
+            materialized='incremental',
+            partition_by='organisation',
+            schema='jaffle_shop',
+            tags='daily'
+        )
+    }}
+    SELECT *
+    FROM {{ ref('my_upstream_model') }}
+    """
+    sql_model.write_text(sql_query)
+    dbt_model = DbtModel("jaffle_shop_revenue", sql_model)
+    dbt_model_config = dbt_model.config
+    assert dbt_model_config.config_selectors == {
+        "materialized:incremental",
+        "schema:jaffle_shop",
+        "tags:daily",
+    }
+
+
+# TODO sources test
```

Comment:
is this a TODO for this PR or a follow-on?

Reply:
Is the testing of sources supported by cosmos?